freetextresponse/freetextresponse.py

```diff
@@ -0,0 +1,437 @@
+'''
+This is the core logic for the FreeText XBlock
+'''
+
+import os
+import re
+
+import pkg_resources
+
+from enum import Enum
+from xblock.core import XBlock
+from xblock.fields import Scope
+from xblock.fields import String
+from xblock.fields import Integer
+from xblock.fields import Float
+from xblock.fragment import Fragment
+
+
+class Freetextresponse(XBlock):
+    #  pylint: disable=too-many-ancestors, too-many-instance-attributes
+    '''
+    Enables instructors to create questions with free-text responses.
+    '''
+    @staticmethod
+    def workbench_scenarios():
+        '''
+        Gather scenarios to be displayed in the workbench
+        '''
+        return [
+            ('FreeText XBlock',
+             '''<sequence_demo>
+                    <freetextresponse />
+                    <freetextresponse name="My First XBlock" />
+                </sequence_demo>
+             '''),
+        ]
+    display_name = String(
+        default='FreeText XBlock',
+        scope=Scope.settings,
+        help="This is the title for this question type",
+    )
+    prompt = String(
+        default='Please enter your response within this text area',
+        scope=Scope.settings,
+        help='This is the prompt students will see when '
```

Comment:
Related: We should look into the work they've done in the xblock-utils library [1], namely creating a default studio view [2].

My work (here: the default studio view) predates the creation of `xblock-utils`. Despite having not explored it thoroughly, it does appear to be a superior implementation.

[edit]...

Upon quick inspection, it _does_ appear to handle help text automagically [3].

Given all of that, we should look to adopt their approach and use their library.

[1] https://github.com/edx/xblock-utils
[2] https://github.com/edx/xblock-utils/blob/08e5a5a9fc8ab46b627435427fd7e04c20809009/xblockutils/studio_editable.py#L69
[3] https://github.com/edx/xblock-utils/blob/08e5a5a9fc8ab46b627435427fd7e04c20809009/xblockutils/studio_editable.py#L117